### PR TITLE
Cart logic tests

### PR DESCRIPTION
**Pull request by KacFlor**

### CODE REVIEW

File: src/main/java/com/KacFlor/ShopSpring/config/UserConfig.java [hunk 1]
```diff
@@ -22,10 +24,13 @@ public UserConfig(){
     }
 
     @Bean
-    CommandLineRunner commandLineRunnerUser(UserRepository userRepository, CustomerRepository customerRepository){
+    CommandLineRunner commandLineRunnerUser(CartRepository cartRepository ,UserRepository userRepository, CustomerRepository customerRepository){
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/config/UserConfig.java:27]: Do you remember as I told you to make special directory for default data creation "developer helper"?

File: src/main/java/com/KacFlor/ShopSpring/service/ProductService.java [hunk 2]
```diff
@@ -161,7 +164,7 @@ public boolean removePromotion(Integer PTid, Integer PNid){
             return true;
         }
         else{
-            throw new ExceptionsConfig.ResourceNotFoundException("Payment not found");
+            throw new ExceptionsConfig.ResourceNotFoundException("Resource not found");
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/service/ProductService.java]: Lets divide it so we there is proper message, when optionalProduct is not present -> Product not found, when optionalPromotion is not present -> Promotion not found, do it as as early return, https://medium.com/swlh/return-early-pattern-3d18a41bba8, apply everywhere

File: src/main/java/com/KacFlor/ShopSpring/config/UserConfig.java (hunk 1 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/config/UserConfig.java:27]: done

File: src/main/java/com/KacFlor/ShopSpring/service/ProductService.java (hunk 2 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/service/ProductService.java]: done

File: src/main/java/com/KacFlor/ShopSpring/config/UserConfig.java (hunk 1 above)
Comment (Ficzer) [reply src/main/java/com/KacFlor/ShopSpring/config/UserConfig.java:27]: It is not done

File: src/main/java/com/KacFlor/ShopSpring/controller/CustomerController.java [hunk 3]
```diff
@@ -72,14 +72,14 @@ public ResponseEntity<?> createCardData(@RequestBody NewCardData newCardData){
 
     @PreAuthorize("hasAnyAuthority('" + Role.Fields.USER + "', '" + Role.Fields.ADMIN + "')")
     @PatchMapping("/me")
-    public ResponseEntity<?> dataChange(@RequestBody NewCustomer newCustomer){
+    public ResponseEntity<?> update(@RequestBody NewCustomer newCustomer){
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/CustomerController.java:75]: Inconsistency

**Review by Ficzer — APPROVED** [review]

**Pull request merged**